### PR TITLE
Remove farm penalty code

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

File: energy-integration/timestamp-oracle/src/epoch_to_timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        self.timestamp_for_epoch(epoch).set(start_timestamp);
+    }
+
+    #[endpoint(updateAndGetTimestampStartEpoch)]
+    fn update_and_get_timestamp_start_epoch(&self) -> Timestamp {
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_update_epoch = self.epoch_last_interaction().get();
+        let mapper = self.timestamp_for_epoch(current_epoch);
+        if current_epoch == last_update_epoch {
+            return mapper.get();
```
Comment (psorinionut) [inline energy-integration/timestamp-oracle/src/epoch_to_timestamp.rs:31]: Here we should always have a value. But because we have external factors, like the owner could update the storage with overwrite, I would also put a check here that the value stored by timestamp_for_epoch is not 0. And I would go even to a require, to completely crash the execution in that particular case, to not negatively impact the rewards computation.

File: energy-integration/farm-boosted-yields/src/custom_reward_logic.rs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+        // Example: user entered at 25% of week, so give only 75% of rewards
+        let enter_timestamp_percent_of_week = linear_interpolation::<Self::Api, _>(
+            week_timestamps.start as u128,
+            week_timestamps.end as u128,
+            claim_progress.enter_timestamp as u128,
+            0,
+            MAX_PERCENT as u128,
+        );
+        let percent_leftover = MAX_PERCENT as u128 - enter_timestamp_percent_of_week;
```
Comment (psorinionut) [inline energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:157]: Why all of this u128 converting?

File: energy-integration/farm-boosted-yields/src/custom_reward_logic.rs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        core::cmp::min(max_rewards, boosted_reward_amount)
+    }
+
+    fn limit_boosted_rewards_by_claim_time(
+        &self,
+        user_reward: BigUint,
+        week_timestamps: &WeekTimestamps,
+        claim_progress: &ClaimProgress<Self::Api>,
+    ) -> BigUint {
+        if !(claim_progress.enter_timestamp >= week_timestamps.start
```
Comment (psorinionut) [inline energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:143]: This is a bit hard to follow, especially negating the check. Could it be made a little bit easier to read?

Comment (psorinionut) [reply energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:143]: The condition makes sense so we can keep it as it is, so we could have something like

```
claim_progress.enter_timestamp < week_timestamps.start
    || claim_progress.enter_timestamp >= week_timestamps.end
```

File: energy-integration/timestamp-oracle/src/epoch_to_timestamp.rs (hunk 1 above)
Comment (psorinionut) [reply energy-integration/timestamp-oracle/src/epoch_to_timestamp.rs:31]: I like the new implementation, where you don't need to terminate the execution, but return clearly that there is no data.

File: energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs [hunk 4]
```diff
@@ -165,3 +184,60 @@ pub trait WeeklyRewardsSplittingModule:
         }
     }
 }
+
+#[cfg(test)]
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs:188]: It's actually common practice to have simple tests like this directly in code, but we haven't adopted it for some reason.

This test is very specific for the struct that's in this file, so I think it's okay for it to remain here.

File: energy-integration/farm-boosted-yields/src/custom_reward_logic.rs (hunk 3 above)
Comment (dorin-iancu) [reply energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:143]: Really? I would've thought it's actually easier to follow this way, since it's just "if not in this range, return".

File: energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (psorinionut) [reply energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs:188]: I'm mixed feelings about this. It may be common practice, but this has nothing to do with the actual contract logic, and I much rather prefer to have as few lines as possible in the actual SC files. It's clear what's what at a first glance.

File: energy-integration/farm-boosted-yields/src/custom_reward_logic.rs (hunk 2 above)
Comment (psorinionut) [reply energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:157]: I don't think we could overflow. We have a maximum number of 6_048_000_000 (max number of seconds in a week * MAX_PERCENT), which is under u64 max limit.

File: energy-integration/farm-boosted-yields/src/custom_reward_logic.rs (hunk 3 above)
Comment (psorinionut) [reply energy-integration/farm-boosted-yields/src/custom_reward_logic.rs:143]: We can keep it like this also, it's ok.

File: energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs:188]: I disagree actually. It's very clearly labeled as a test and it's in its own module too, so it's not polluting the implementation in any way. Maybe if it was at the beginning of the file I'd agree.

**Review by psorinionut — COMMENTED** [review]
Looks good for me, but I think we should discuss how we organize the features branches. I'll green it after I take a look at all pending features and where we direct them.

**Review by psorinionut — APPROVED** [review]

**Comment by psorinionut** [comment]
Should not be merged until we have a clear launch date.
